docs/user/config-file/v2.rst

```diff
@@ -456,6 +456,43 @@ See :doc:`/build-customization` for more details.
 :Default: ``{}``
 
 
+build.commands
+``````````````
+
+Specify a list of commands that Read the Docs will run on the build process.
+When ``build.commands`` is used, none of the :term:`pre-defined build jobs` will be executed.
+(see :doc:`/build-customization` for more details).
+This allows you to run custom commands and control the build process completely.
+The ``output/`` directory (relative to the checkout's path) will be uploaded and hosted by Read the Docs.
+
+.. warning::
+
+   This feature is in a *beta phase* and could suffer incompatible changes or even removed completely in the near feature.
+   It does not yet support some of the Read the Docs' integrations like the :term:`flyout menu`, search and ads.
+   However, integrating all of them is part of the plan.
+   Use it under your own responsibility.
+
+.. code-block:: yaml
+
+   version: 2
+
+   build:
+     os: ubuntu-22.04
+     tools:
+       python: "3.10"
+     commands:
+       - pip install pelican
+       - pelican --settings docs/pelicanconf.py --output output/ docs/
+
+.. note::
+
+   ``build.os`` and ``build.tools`` are also required when using ``build.commands``.
```

Comment:
Yea, feels like build.tools being required here is a bit odd? If I'm using a Hugo, I don't need any of our supported tools..

Reply:
I was confused by this with `build.jobs` as well.

Reply:
Yeah, I mentioned this in the notes of the PR.

> We should probably make build.tools not required since it's possible to generate HTML without any of these languages as shown in the first example where only echo command is used

Should we make `build.tools` _not required_ on this initial version? That will require some work on the config file because it implemented making `.tools` the required one instead of `.os` to decide between `build.image` and `build.os`.